stac_fastapi/api/stac_fastapi/api/app.py

```diff
@@ -83,9 +86,15 @@ class StacApi:
     api_version: str = attr.ib(default="0.1")
     stac_version: str = attr.ib(default=STAC_VERSION)
     description: str = attr.ib(default="stac-fastapi")
+    search_get_request_base_model: Type[BaseSearchGetRequest] = attr.ib(
```

Comment:
Yeah, this is a potentially ugly and maybe a point where you could provide some counsel. Basically, the need to supply a catalog URI parameter changes the constructed search model  so what I'm doing here is passing in the base `search_get_request_base_model` because it might possibly differ from the fully constructed `search_get_request_model` used by standard searches. Perhaps the answer is to just construct both of these inside of `core.py`. Let me know what you think.